homeassistant/components/myq/__init__.py

```diff
@@ -63,3 +73,42 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
         hass.data[DOMAIN].pop(entry.entry_id)
 
     return unload_ok
+
+
+class MyQEntity(CoordinatorEntity):
+    """Base class for MyQ Entities."""
+
+    def __init__(self, coordinator: DataUpdateCoordinator, device) -> None:
+        """Initialize class."""
+        super().__init__(coordinator)
+        self._device = device
+        self._attr_unique_id = device.device_id
+
+    @property
+    def name(self):
+        """Return the name if any, name can change if user changes it within MyQ."""
+        return self._device.name
```

Comment:
```suggestion
        self._attr_unique_id = device.device_id
        self._attr_name = device.name

```

Reply:
If I do this then the entity_id ends up being <domain>.unnamed_device

Reply:
Also, device_id will never change as that is the serial number

Reply:
Why does the name change?

Reply:
So the name is coming from MyQ. Within their app one can change the name and when that is done the pymyq API will thus get the new name.
Having it non-static will then also result in the name being updated accordingly within HASS.